src/completion-provider/template-completion-provider.ts

```diff
@@ -53,33 +54,79 @@ export default class TemplateCompletionProvider {
     let completions: CompletionItem[] = [];
 
     if (isMustachePath(focusPath)) {
-      completions.push(...listComponents(project.fileIndex));
-      completions.push(...listHelpers(project.fileIndex));
+      completions.push(...listComponents(project));
+      completions.push(...listHelpers(project));
       completions.push(...emberMustacheItems);
     } else if (isBlockPath(focusPath)) {
-      completions.push(...listComponents(project.fileIndex));
+      completions.push(...listComponents(project));
       completions.push(...emberBlockItems);
     } else if (isSubExpressionPath(focusPath)) {
-      completions.push(...listHelpers(project.fileIndex));
+      completions.push(...listHelpers(project));
       completions.push(...emberSubExpressionItems);
     } else if (isLinkToTarget(focusPath)) {
-      completions.push(...listRoutes(project.fileIndex));
+      completions.push(...listRoutes(project));
     }
 
     return filter(completions, getTextPrefix(focusPath), { key: 'label' });
   }
 }
 
-function listComponents(index: FileIndex): CompletionItem[] {
-  return uniqueBy(index.files.filter(isComponent), 'name').map(toCompletionItem);
-}
-
-function listHelpers(index: FileIndex): CompletionItem[] {
-  return index.files.filter(isHelper).map(toCompletionItem);
-}
-
-function listRoutes(index: FileIndex): CompletionItem[] {
-  return index.files.filter(isRoute).map(toRouteCompletionItem);
+function listComponents(project: Project): CompletionItem[] {
+  const { root } = project;
+  const jsPaths = walkSync(join(root, 'app', 'components'));
+  const hbsPaths = walkSync(join(root, 'app', 'templates', 'components'));
```

Comment:
I'm wondering if we should move to async fs walking 🤔

Reply:
I think async walking would require a bigger refactor. Not sure if it is worth the effort since the LS is running in it's own process anyway so it is not blocking anything else.